src/amg/azext_amg/migrate.py

```diff
@@ -0,0 +1,302 @@
+from knack.log import get_logger
+from azure.cli.core.style import print_styled_text, Style
+
+from .restore import (create_dashboard, create_folder, create_library_panel, create_snapshot, create_annotation,
+                      create_datasource, set_uid_mapping, check_folder_exists, check_library_panel_exists,
+                      check_dashboard_exists, check_snapshot_exists, check_annotation_exists_and_return_id)
+from .backup_core import (get_all_dashboards, get_all_library_panels, get_all_snapshots, get_all_folders,
+                          get_all_annotations, get_all_datasources)
+
+logger = get_logger(__name__)
+
+
+def migrate(backup_url, backup_headers, restore_url, restore_headers, dry_run,
+            overwrite, folders_to_include=None, folders_to_exclude=None):
+    # pylint: disable=too-many-locals
+
+    # get all datasources to be backedup
+    all_datasources = get_all_datasources(backup_url, backup_headers)
+    all_restore_datasources = get_all_datasources(restore_url, restore_headers)
+    if (all_restore_datasources is None) or (all_datasources is None):
+        logger.error("ABORTING!! Datasources are not found. Please check the URLs, headers, or api key/service token.")
```

Comment:
Nope, I guess the error message wasn't good enough. The issue is mainly when the source / restore url doesn't seem to work, so that the response isn't 200 and thus the function will return None.

Reply:
do you mean if the get_all_datasources request fails?

Reply:
get_all_datasources can fail since we don't do a health check on the restore url (only on the backup/destination url), I probably shouldn't have the logic here but in the main migrate function.

Reply:
Here is a better fix: https://github.com/Azure/azure-cli-extensions/pull/7802/commits/f2e2f43ce7538c3f01a415a541a980de93a00b9f